### DISCUSSION TITLE
Looking for new maintainer

### DISCUSSION THREAD

**Discussion by campbellwray** [silentpartnersoftware/Keycloak.Net#10] [General]
Hi,

My company might be interested in taking over maintenance of this project, we are a small team of about 5 developers and we are currently rolling out Keycloak in our tech stack. Our apps are primarily .NET so we are using your library to interact with Keycloak, and intend to do so for the foreseeable future.

I have talked to management at my company as we will need to support this library going forward anyway and that it would be good to give back to the community with any enhancements we make. They are interested in the idea, but are first requesting some more information as to what sort of time commitment should be expected for this project.

If I had to guess I would say that it would be a few hours per month reviewing PRs and creating new releases on NuGet. Would you be able to provide some further insight into what would be involved in maintaining this project?

Thanks,
Campbell

**Comment by campbellwray** [comment]
@AnderssonPeter tagging you for visibility just in case you don't have notifications enabled

**Reply by AnderssonPeter** [reply]
Your right i don't have notifications enabled for this project don't know why.

Do you have a user that would be the owner? I would like to check what you currently have before transferring.

**Reply by AnderssonPeter** [reply]
Sadly i more or less only accept pr's and create releases. The release pipeline is more or less automated if i remember correctly.

So doing that is ready, what i can't tell you it's how much fixing is needed for keeping it up to date as i haven't invested that time.

Also i would update and automate the unit tests as i haven't checked them at all.

**Reply by campbellwray** [reply]
Unfortunately all of the codebases that I have worked on recently are in my companies private Bitbucket workspace, so you won't be able to view that, but I will check with the others on my team to see if anyone has anything more substantial.

I did make a [PR to this repository](https://github.com/AnderssonPeter/Keycloak.Net/pull/9) a few weeks back, so you're welcome to look at my Github profile, but I realise that doesn't really demonstrate much.

We have been using this library with Keycloak 20.x and it seems to be pretty stable still, other than there being a few properties missing here and there, so I suspect that there isn't much that needs to be fixed. As for the unit tests, I believe I tried to run the tests but I couldn't get them working, but if they are known to have issues then that shouldn't be a problem to sort out

**Reply by campbellwray** [reply]
This is the company website, unfortunately no code here though - [https://www.silentpartnersoftware.com/](https://www.silentpartnersoftware.com/)

**Reply by AnderssonPeter** [reply]
I'm fine with transferring both the repo and the nugget, get back to me if your company is fine with it.

**Reply by campbellwray** [reply]
Thanks, I will get back to you when I have an answer.

**Reply by campbellwray** [reply]
@AnderssonPeter I have got approval from management to be set up as the maintainer for this project, can you please transfer ownership to this GitHub account (@campbellwray) and to the NuGet user sps_campbellwray. 

If you have any issues please let me know

**Reply by AnderssonPeter** [reply]
@campbellwray I can't transfer the project while you have a github repo named Keycloak.Net

**Reply by campbellwray** [reply]
@AnderssonPeter I have removed the repo, you should be able to transfer now

**Reply by AnderssonPeter** [reply]
Sorry for the delay, but the github repo has been transfer or at least a request for transfer has been created.

**Reply by AnderssonPeter** [reply]
And now you have a pending approval for the nuget :)

**Reply by campbellwray** [reply]
Thanks, I have approved them both. Do you have any documentation or anything that might be helpful for maintaining the repository or creating releases? We mostly use Bitbucket, so I am not super familiar with Github, but if not I am sure we will work it out.

**Reply by AnderssonPeter** [reply]
What I think needs to be done:
1. Upgrade furl to latest version #6 
2. See if there are any other libraries that are out of date
3. Optional, but using github actions to create your nuget package and upload it to nuget.org would be nice.
